Source/WebCore/accessibility/AccessibilityScrollView.cpp

```diff
@@ -53,6 +53,33 @@ AccessibilityScrollView::~AccessibilityScrollView()
     ASSERT(isDetached());
 }
 
+bool AccessibilityScrollView::isRoot() const
+{
+    if (m_frameOwnerElement) {
+        // An owned scrollview (e.g. belonging to an iframe) can never be the root.
+        return false;
+    }
+    // Though keep in mind, m_frameOwnerElement is a WeakPtr, so the lack of its presence alone
+    // doesn't guarantee that this is the root, as the element may have been destroyed but |this|
+    // is still valid and undestroyed for some amount of time.
+
+    return m_scrollView && !m_scrollView->parent();
```

Comment:
If we're on the main thread, I feel like we should be able to get a 100% correct answer here, rather than using a heuristic that has a chance of being wrong under other weird circumstances.

For example, or m_scrollView is a LocalFrameView, which has a LocalFrame, which has a localMainFrame - if it's not the main frame then it's not the root.

Or alternatively if we cache AXObjectCache when constructing AccessibilityScrollView, then use the Document from AXObjectCache and go to Document > Frame > FrameView and see if it's the same as m_scrollView - if not, it's not the root.